### PR TITLE
Fixes #623 the Dim.Percent.

### LINKED ISSUE #623: Dim.Percent() generating incorrect Width
The code below produces this result. The right hand TextField does not have a width of 40%

image

```csharp
Application.Init();

var win = new Window("Test") { X = 0, Y = 0, Width = Dim.Fill(), Height = Dim.Fill() };

var textLeft = new TextField("Left") { X = 0, Y = 0, Width = Dim.Percent(40) };
var textRight = new TextField("Right") { X = Pos.Right(textLeft)+1,  Width = Dim.Percent(40) };
win.Add(textLeft, textRight);

Application.Top.Add(win);
Application.Run ();
```

### PR DESCRIPTION
**Pull request by BDisp**

### CODE REVIEW

**Comment by tig** [comment]
As I noted here, we need @migueldeicaza to approve this.

https://github.com/migueldeicaza/gui.cs/issues/623

**Comment by BDisp** [comment]
Okay, I just presented my solution.

**Comment by BDisp** [comment]
This isn't already fixed in Keys.cs?

			// ProcessHotKey log:
			// BUGBUG: Label is not positioning right with Pos, so using TextField instead


			// ProcessColdKey log:
			// BUGBUG: Label is not positioning right with Pos, so using TextField instead

**Comment by migueldeicaza** [comment]
I left some comments on the original issue, because the way that I see it, this runs the spectrum from "poor documentation" to "poor behavior", and would love to get more thoughts on this.

**Comment by BDisp** [comment]
> 
> 
> I left some comments on the original issue, because the way that I see it, this runs the spectrum from "poor documentation" to "poor behavior", and would love to get more thoughts on this.

@migueldeicaza the solution I proposing covers both `DimFactor `behaviors.

**Comment by migueldeicaza** [comment]
I see it now, the "r" parameter, let us go ahead and merge this.

File: Terminal.Gui/Types/Rect.cs [hunk 1]
```diff
@@ -17,6 +17,9 @@ namespace Terminal.Gui
 	/// </summary>
 	public struct Rect
 	{
+		int width;
```
[the remark this answers is not in the document]
Comment (BDisp) [reply Terminal.Gui/Types/Rect.cs:20]: That is up to you. I'm going to remove them and it's over.

File: Terminal.Gui/Core/PosDim.cs [hunk 2]
```diff
@@ -407,12 +415,12 @@ public override string ToString ()
 		/// };
 		/// </code>
 		/// </example>
-		public static Dim Percent (float n)
+		public static Dim Percent (float n, bool r = false)
```
[the remark this answers is not in the document]
Comment (BDisp) [reply Terminal.Gui/Core/PosDim.cs:418]: As 'n' is also not. This has already been approved by @migueldeicaza and does not justify complicating matters. So I don't change it. Perhaps you should continue on this project alone.

File: Terminal.Gui/Types/Rect.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (migueldeicaza) [reply Terminal.Gui/Types/Rect.cs]: We won’t be using System.Drawing types.

**Pull request closed without merging**